linera-client/src/chain_listener.rs

```diff
@@ -218,7 +218,6 @@ impl<C: ClientContext> ChainListener<C> {
             }
             Reason::NewRound { .. } => self.update_validators(&notification).await?,
             Reason::NewBlock { hash, .. } => {
-                self.update_validators(&notification).await?;
```

Comment:
We are processing notifications from the local node here. This could be about a certificate we have just put together locally, from the validator signatures (i.e. _no_ validator has it yet); and even if we received it from a validators, we only know that at least _one_ of them has it.

I can't easily come up with a test scenario where this would break, off the top of my head, but I also don't see a reason why this call would always be unnecessary.

Reply:
We sent `ConfirmedBlock` to all validators already though (or will send) so they know about it. I'm not sure if we need to notify them _again_ every time one of validators accepts the new block.

Reply:
Did we? Even if we only processed it because _one_ validator notified us about the new block?

Reply:
They may each have _signed_ the certificate, but they may not have the actual certificate with a quorum of signatures yet.

E.g. if a faulty client sent the final certificate (with a quorum of signatures) only to _one_ validator, the others don't have it, even though they signed it. Our own client may now receive the notification from that one validator, download the certificate from it and handle it locally. That would trigger this notification.

Reply:
The only way we could have constructed the `ConfirmedBlockCertificate` is if the quorum of validators responded to our `ConfirmedBlock`. It could have happened if we
- sent the fast block
- went through the `FinalizeBlock` logic

In both cases we communicate with all validators to send them the confirmed certificate. If they are online and responsive they will get it. If they're not, I don't see how calling it again here would be making a difference.

Reply:
(Anyway, this is not a blocker from my side!)